linalg/ginkgo.hpp

```diff
@@ -45,6 +45,12 @@ template <typename T> using gko_array = gko::Array<T>;
 template <typename T> using gko_array = gko::array<T>;
 #endif
 
+#if MFEM_GINKGO_VERSION < 10600
+#define GKO_LEND(x) gko::lend(x)
```

Comment:
My first reaction here is that we should avoid `#define` in headers which are installed by MFEM like this one since it will affect all downstream user code. We could move some of these function definitions and the macro into `ginkgo.cpp` to fix this. But on second thought maybe these are safe, and moving things into the `.cpp` will be more trouble than necessary. Maybe @v-dobrev do you have an opinion here?

For the template functions you may just need to explicitly instantiate them in the `.cpp` file since they aren't being defined in the same file where they are declared. For example:

```
template double Ginkgo::compute_norm(const gko::matrix::Dense<double> *);
```

Reply:
I could also just put two separate lines of code with the compiler guards (one line with the `gko::lend`, one without). It's a bit messier to read, but it would avoid any `#define`s. Would that be better?

Reply:
That seems fine. I was trying to decide if we could avoid defining `MFEM_GINKGO_VERSION` in the header too but this one is probably fine and is based on values from the Ginkgo headers rather than other compile time definitions. So I think your solution would be OK!

Reply:
I think having macros in MFEM headers is fine, however, I'd add a prefix `MFEM_` (`GKO_LEND` -> `MFEM_GKO_LEND`) since it is a macro defined by MFEM. Of course, removing the macro is also fine, if there is a preference for that approach -- I'm fine with either choice.

Reply:
Based on discussion here, I went ahead and removed the use of `GKO_LEND` but left `MFEM_GINKGO_VERSION` in ginkgo.hpp.